scio-extra/src/main/scala/com/spotify/scio/extra/sparkey/SparkeyIO.scala

```diff
@@ -0,0 +1,56 @@
+package com.spotify.scio.extra.sparkey
+
+import java.util
+
+import com.spotify.scio.ScioContext
+import com.spotify.scio.io.{EmptyTap, EmptyTapOf, ScioIO, Tap, TapOf}
+import com.spotify.scio.util.RemoteFileUtil
+import com.spotify.scio.values.SCollection
+import com.spotify.sparkey.{IndexHeader, LogHeader, SparkeyReader}
+import org.apache.beam.sdk.options.PipelineOptions
+
+final case class SparkeyUriTap(private val singleSparkeyUri: SparkeyUri) extends Tap[SparkeyUri] {
+  override def value: Iterator[SparkeyUri] = List(singleSparkeyUri).toIterator
+  override def open(sc: ScioContext): SCollection[SparkeyUri] = sc.parallelize(value.toIterable)
+}
+
+object SparkeyIO {
+  object ReadP {
+    def apply(options: PipelineOptions): ReadP = ReadP(() => RemoteFileUtil.create(options))
+  }
+  case class ReadP(rfuCreateFn: () => RemoteFileUtil)
+
+  def testUri(in: List[(String, String)]): SparkeyUri = {
```

Comment:
`in` can just be a `Map`? Also `SparkeySideInput.forTest(data)` if we rename the class & the method?
Also we should overload this to support `Array[Byte] => Array[Byte]` too, but you'll have to use something like `ByteString` internally to make map lookups work.